File: python3/utf8.py
#! /usr/local/bin/python3

import regex

output = open("output.txt","w")
with open("../utf8.txt",encoding="utf-8",mode="r") as exampleFile:
    for line in exampleFile:
        # Has new line from file input ensure we write out the proper encoded version
        output.write("String: "+line)
        print("String: "+line[:-1])

        output.write("String type: "+str(type(line))+"\n")
        print("String type: "+str(type(line)))

        # Note that if you preface a string with u it becomes unicode instead of ascii
        strList = u", ".join(list(line))
        output.write("List: " + strList)
        print("List: " + strList[:-1])

        output.write("Length: "+str(len(line))+"\n")
        print("Length: "+str(len(line)))

        output.write("Seventh character: "+ line[6]+"\n")
        print("Seventh character: " + line[6])

        graphemeArr = regex.findall(r'\X', line, regex.U)
        graphemes = u", ".join(graphemeArr)
        output.write("Grapheme(s): " + graphemes + "\n\n")
        print("Grapheme(s): " + graphemes + "\n")
output.close()
exampleFile.close()

print("This output is also available in output.txt")

File: nodejs/utf8.js
const fs = require('fs');

let result = fs.readFileSync('../utf8.txt', {encoding: 'utf-8'});
let delim = result.indexOf("\r\n") >= 0 ? "\r\n" : "\n";

result = result.split(delim);
var content = ""
for (let i=0; i<result.length; i++){
    line = result[i];
    if (line){
        let grapheme = [...new Intl.Segmenter().segment(line)];
        let joinedGSegments = grapheme.map(seg => {
            return seg.segment;
        })
        content += "String: " + line
        content += "\nString type: " + typeof(line);
        content += "\nList: " + line.split('');
        content += "\nSeventh Character: " + line[6]
        content += "\nGrapheme(s)" + joinedGSegments.join(', ');
        content += "\nLength: " + line.length + "\nList Length: " + line.split('').length + "\nGrapheme(s) length: " + grapheme.length + "\n\n";
    }
}
console.log(content);
fs.writeFileSync('./output.txt', content);

